### PR TITLE
[Term Entry] JavaScript Array.prototype.toSorted() - close #2650

### LINKED ISSUE #2650: [Term Entry] JavaScript Array.prototype.toSorted()
### Reason/inspiration (optional)

Add the new ES2023 JavaScript array feature, .toSorted(). Open for anyone to tackle this issue!

Reference [Ecma TC39 Repo](https://github.com/tc39/proposal-change-array-by-copy) and [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted).

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Nostromos**
### Description

Closes #2650.

Adding a new folder and term entry for `toSorted()` - a non-destructive JS array method which takes an array and returns a new array in unicode code point ascending order. 

Motivated to do this for a few reasons: 

- complete a codecademy course that requires contributing to docs
- useful method that is new to the language as of this year
- good learning opportunity for array methods and unicode code point
- opportunity to practice working with git & github

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (dakshdeepHERE) [inline content/javascript/concepts/arrays/terms/toSorted/toSorted.md:92]:
```suggestion
console.log('Numbers Ascending: ' + ascending); 
console.log('Numbers Descending: ' + descending); 
```
There's no need to add comment about the output as the viewer can run these code blocks themselves

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few changes from my side

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md]: Updates made in latest commit.

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md]: Updates made in latest commit.

**Comment by Nostromos** [comment]
@dakshdeepHERE All your suggestions accepted and updated per the most recent commit.

**Comment by dakshdeepHERE** [comment]
Awsm @Nostromos I have one more doubt idk if attaching wiki link is allowed. Let's wait for the maintainer check to confirm once.

**Comment by Nostromos** [comment]
Great catch. Outside links are forbidden except to standards docs or entities. I'd meant to change that link to point to: https://developer.mozilla.org/en-US/docs/Glossary/Code_point

However, I accidentally lost that change when I ran the wrong git commands. I'll make that change now. I will note that sort.md also includes that link but I won't make that change to keep this PR scoped.

**Comment by Nostromos** [comment]
@caupolicandiaz Ready for your review.

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/toSorted/toSorted.md:19]:
```suggestion
The **`.toSorted()`** method takes an array and returns a new array with the elements sorted in ascending order. It does not mutate the original array. All undefined elements are sorted to the end of the array.
```

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/toSorted/toSorted.md:21]:
```suggestion
The sorting is done by converting array elements to [strings](https://www.codecademy.com/resources/docs/javascript/strings) and comparing them by Unicode code point value. 
```

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/toSorted/toSorted.md:25]: I think it may be preferable to combine all of these in one code block and the text in bold can be added to the right of the syntax as a comment

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/toSorted/toSorted.md:55]:
```suggestion
In the example below, the `.toSorted()` method is applied to the arrays `sports` and `numbers`.
```

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/toSorted/toSorted.md:66]: add an empty line

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/toSorted/toSorted.md:80]: This function is not currently supported by the Codecademy infrastructure so the codebyte will throw an error. Please remove this example.

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@Nostromos I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md:19]: 🤦‍♂️ Good catch.

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md:25]: @caupolicandiaz I initially did that but with the code in pseudo and the few lines of additional explanation, it was a little hard to parse. Is that a requirement for you to approve?

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md:80]: Ugh you're totally right. I forgot to check what version of node it was running. 

Any opposition to moving that to the example section? I think most people hitting the article will want to sort numerically and the method doesn't behave intuitively for sorting numbers numerically.

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md:25]: N/m I'll make it work.

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md]: Fixed in #[5a59c46](https://github.com/Codecademy/docs/pull/2716/commits/5a59c46fe45eb19a8b49792b070baee48386a5a4).

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md:65]: ✅ in [5a59c46](https://github.com/Codecademy/docs/pull/2716/commits/5a59c46fe45eb19a8b49792b070baee48386a5a4).

File: content/javascript/concepts/arrays/terms/toSorted/toSorted.md
Comment (Nostromos) [reply content/javascript/concepts/arrays/terms/toSorted/toSorted.md]: ✅ in [5a59c46](https://github.com/Codecademy/docs/pull/2716/commits/5a59c46fe45eb19a8b49792b070baee48386a5a4).

**Comment by Nostromos** [comment]
@caupolicandiaz Changes made per your request. Ready for your review. Thank you.

**Review by caupolicandiaz — APPROVED** [review]
@Nostromos I pushed a couple edits, mostly to update spacing. Looks good for a second review.

**Comment by Nostromos** [comment]
Thank you @caupolicandiaz!

**Review by SSwiniarski — APPROVED** [review]
@Nostromos looks good to go.

**Comment by SSwiniarski** [comment]
@Nostromos Congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/javascript/arrays/toSorted

image

You're currently anonymous. If you don't wish to be anonymous, check out this forum thread on how to link your GitHub and Codecademy profiles: https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036  **Note:** it may take a few minutes for the change to register on the entry page.

**Comment by Nostromos** [comment]
Thank you @SSwiniarski!

**Pull request merged**
